NGitLab.Tests/Docker/GitLabTestContextRequestOptions.cs

```diff
@@ -33,86 +35,102 @@ public override WebResponse GetResponse(HttpWebRequest request)
             }
 
             WebResponse response = null;
+
+            // GitLab is unstable, so let's make sure we don't overload it with many concurrent requests
+            s_semaphoreSlim.Wait();
             try
             {
-                response = base.GetResponse(request);
-            }
-            catch (WebException exception)
-            {
-                response = exception.Response;
-                if (response is HttpWebResponse webResponse)
+                try
```

Comment:
Do we really need the nested try/catch/finally blocks? Could we not simply release the semaphore from the same finally block as before, after the call to `response = LogRequest(request, response)`?

Reply:
LogRequest is using the response stream, so it could fail. I think it's safer to keep it in an outer try/finally block

Reply:
...unless of course the said call can actually throw.

Reply:
Oops, forgot to refresh before posting my last comment. 🙄